bazel_integration_test/script_test.bzl

```diff
@@ -0,0 +1,116 @@
+"A facility for script tests that create workspaces and call Bazel."
+
+load(
+    "//bazel_integration_test:defs.bzl",
+    "bazel_integration_test",
+)
+
+_RUNNER_SCRIPT = """
+#!/bin/bash
+
+script="%s"
+
+working_dir="."
+cd "${working_dir}"
+
+export bazel
+
+# Run the actual tests
+# Bazel is available as ${BIT_BAZEL_BINARY}.
+if "${script}"; then
+  echo "test passed"
+else
+  echo "test failed"
+  exit 1
+fi
+
+exit 0
+"""
+
+def _write_runner_impl(ctx):
+    script = ctx.file.script
+    runner = ctx.outputs.out
+
+    ctx.actions.write(
+        output = runner,
+        content = _RUNNER_SCRIPT % script.path,
+    )
+
+    return [DefaultInfo(
+        executable = runner,
+        runfiles = ctx.runfiles(files = ctx.files.script),
+    )]
+
+_write_runner = rule(
+    implementation = _write_runner_impl,
+    attrs = {
+        "out": attr.output(mandatory = True),
+        "script": attr.label(allow_single_file = True, mandatory = True),
+    },
+    executable = True,
+)
+
+def script_test(
+        name,
+        srcs,
+        deps,
+        bazel_binaries,
+        bazel_version):
+    """
+    Macro that allows writing script tests which create a Bazel workspace and call the binary directly.
+
+    These are intended to be like the Bazel integration tests found at https://github.com/bazelbuild/bazel/tree/master/src/test/shell/integration.
+
+    Tests will find the Bazel binary via the `${BIT_BAZEL_BINARY}` environment variable.
+
+    Args:
+        name: name of the resulting test
+        srcs: The file containing the shell script. This attribute must be a
```

Comment:
Are you calling it `srcs` instead of `src` because of standard naming conventions? Since this is a macro, I would be fine with `src`. Your call.

Reply:
Yeah, I copied the text from sh_test. I'm going to leave this as-is.